addons/project/views/project_views.xml

```diff
@@ -812,7 +818,8 @@
                 <kanban default_group_by="stage_id" class="o_kanban_small_column o_kanban_project_tasks" on_create="quick_create" quick_create_view="project.quick_create_task_form" examples="project" js_class="project_kanban" sample="1">
                     <field name="color"/>
                     <field name="priority"/>
-                    <field name="stage_id" options='{"group_by_tooltip": {"description": "Description"}}'/>
+                    <field name="stage_id" options='{"group_by_tooltip": {"description": "Description"}}' groups="project.group_project_user"/>
+                    <field name="stage_id" options='{"group_by_tooltip": {"description": "Description"}}' readonly="1" groups="!project.group_project_user"/>
```

Comment:
I don't think it is needed to do that since the internal user without any access to project app, cannot see the kanban view of tasks. IMO, he can only see the form view, right?